kedro-airflow/kedro_airflow/grouping.py

```diff
@@ -62,7 +63,7 @@ def group_memory_nodes(catalog: DataCatalog, pipeline: Pipeline):
                         sequence_id = sequence_map[i]
 
                 # Append to map
-                node_sequences[sequence_id].append(node)
+                node_sequences[sequence_id].append(node)  # type: ignore[index]
```

Comment:
It flags for the case that we look at an input that is not a memory dataset. An `if sequence_id:` condition should assuage mypy. Logically we know that because some input must be a memory dataset, this line will never throw an error - I think the if condition makes this logic more clear than ignoring the mypy warning. What do you think @ankatiyar ?

Reply:
I tried that and then `ruff` throws - `PLR0912 Too many branches (13 > 12)` so it's either silence `mypy` error or `ruff` 🤔

Reply:
Or can we break up this method in a way?

Reply:
Leaving this in for now, will try to do a follow up for trying to remove ignore statements once `datasets` is also done

Reply:
Don't forget to create an issue for the follow up 😄